scripts/tool_discovery_benchmark/run_benchmark.py

```
@@ -0,0 +1,794 @@
+    if not path.exists():
+        return []
+    by_id: dict[str, dict[str, Any]] = {}
+    for line_number, line in enumerate(path.read_text(encoding="utf-8").splitlines(), 1):
+        if not line.strip():
+            continue
+        observation = json.loads(line)
+        observation_id = observation.get("observation_id")
+        if not isinstance(observation_id, str) or not observation_id:
+            raise ValueError(
+                f"{path}:{line_number} is missing a non-empty observation_id"
+            )
```

Ruff (0.16.1)

[warning] Avoid specifying long messages outside the exception class

(TRY003)